pkg/controller/cloudmetrics/cloudmetrics_controller.go

```diff
@@ -77,47 +78,95 @@ var _ reconcile.Reconciler = &ReconcileCloudMetrics{}
 
 // ReconcileCloudMetrics reconciles a CloudMetrics object
 type ReconcileCloudMetrics struct {
-	client client.Client
-	scheme *runtime.Scheme
-	logger *logrus.Entry
+	client               client.Client
+	scheme               *runtime.Scheme
+	logger               *logrus.Entry
+	postgresProviderList []providers.PostgresMetricsProvider
+	redisProviderList    []providers.RedisMetricsProvider
 }
 
 // Reconcile reads all redis and postgres crs periodically and reconcile metrics for these
 // resources.
 // The Controller will requeue the Request every 5 minutes constantly when RequeueAfter is set
 func (r *ReconcileCloudMetrics) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	r.logger.Info("Reconciling CloudMetrics")
+	r.logger.Info("reconciling CloudMetrics")
+	ctx := context.TODO()
```

Comment:
Let's use `context.Background()`, it's effectively the same but shows the intention that we aren't going to change it any time soon

Reply:
:+1: